src/test/java/teammates/test/cases/logic/EmailsTest.java

```diff
@@ -59,34 +64,44 @@ public static void classSetUp() throws Exception {
 
     @BeforeMethod
     public void caseSetUp() throws ServletException, IOException {
+        /* For Sendgrid testing */
+        fromEmail = "Admin@" + Config.inst().getAppId() + ".appspotmail.com";
+        fromName = "TEAMMATES Admin";
+        replyTo = "teammates@comp.nus.edu.sg";
         
-        InternetAddress internetAddress = new InternetAddress("Admin@"
-                + Config.inst().getAppId() + ".appspotmail.com",
-                "TEAMMATES Admin");
+        /* For MimeMessage testing */
+        InternetAddress internetAddress = new InternetAddress(fromEmail, fromName);
         from = internetAddress.toString();
-        replyTo = "teammates@comp.nus.edu.sg";
     }
 
     @Test
     public void testGetEmailInfo() throws MessagingException {
-
-        Session session = Session.getDefaultInstance(new Properties(), null);
-        MimeMessage message = new MimeMessage(session);
-
         String email = "receiver@gmail.tmt";
         String from = "sender@gmail.tmt";
-
-        message.addRecipient(Message.RecipientType.TO, new InternetAddress(
-                email));
-
-        message.setFrom(new InternetAddress(from));
         String subject = "email subject";
-        message.setSubject(subject);
-        message.setContent("<h1>email body</h1>", "text/html");
 
-        assertEquals(
-                "[Email sent]to=receiver@gmail.tmt|from=sender@gmail.tmt|subject=email subject",
-                Emails.getEmailInfo(message));
+        if (Config.isUsingSendgrid()) {
```

Comment:
I am testing both ways - with and without Sendgrid. The if-else is used because the API for setting email attributes to the expected message is different. Also, the if and else call an overloaded method getEmailInfo(Sendgrid) and getEmailInfo(MimeMessage).